### ISSUE TITLE
Thumbnail generation to slow with only 1 process run in background.

### ISSUE THREAD

**Issue by GokuHiki** [f3d-app/f3d#652]
**Context**
F3D thumbnailer implementation on Windows rely on an old Windows standard which requires creating a F3D process every time, which is slow.
There is a new standard for thumbnails on Windows that rely on sandboxing and would be much more efficient. We should use that standard.

**Suggested solution**
 - https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ee330724(v=vs.85)?redirectedfrom=MSDN
 - https://learn.microsoft.com/en-us/previous-versions//bb776855(v=vs.85)?redirectedfrom=MSDN

**Skills Needed**
 - Windows C++

**Notes**
Original issue:

**Is your feature request related to a problem? Please describe.**
Old F3D shell thumbnails generation can start many instances run in background but the new F3D only run 1 process in background. When I open folder look for an obj file mesh, I must wait for F3D generate thumbnails. I have a few hundred small model files obj in 1 folder and F3D takes too much time to generate thumbnails when my CPU cores bare use at all. My obj files are small and it should be fine to run a lot more processes.

**Describe the solution you'd like**
I would like that F3D can spawn more processes in background when generating thumbnails like before.

**Describe alternatives you've considered**
It would be best if I can config how many processes F3D can run in the background.

**Additional context**
I run F3D night build on windows OS.

**Comment by mwestphal** [comment]
Which version were working fine and which version are not working anymore?

**Comment by mwestphal** [comment]
@GokuHiki ?

**Comment by GokuHiki** [comment]
I think it begins with v1.3.
As before when I open a new folder with a lot of models file F3D uses a lot of CPU but it generates a lot of thumbnails at one so it fast.
But now it only generates one file after finished a file before.

**Comment by mwestphal** [comment]
Then this may be caused by https://github.com/f3d-app/f3d/pull/137 which was merged for 1.2.1.

We had no choice but to reduce performance to avoid potentially completely crashing windows explorer with the previous (but more efficient) technique.

Ideally, thumbnail support should be implemented using the modern sandboxing technique which should be efficient AND safe.

@jpouderoux @Meakk

**Comment by GokuHiki** [comment]
Yes. In version 1.2.1, F3D rendering of thumbnails has become so slow!
It would be nice if there could be an option to re-enable the old thumbnail generate mode in the config file!
Thanks, and regards!

**Comment by mwestphal** [comment]
related to https://github.com/f3d-app/f3d/issues/777

**Comment by mwestphal** [comment]
superseeded by https://github.com/f3d-app/f3d/issues/797

**Comment by mwestphal** [comment]
Should have been closed.
